### ISSUE TITLE
FR: New Mission File Templates

### ISSUE THREAD

**Issue by wookieejedi** [scp-fs2open/fs2open.github.com#3235]
Background:
When you create a new mission it asks you to `Save As` when you click save. Having `Save As` be the default option for the first opening of a new mission is useful so you are not accidentally overwriting another mission when clicking save. I can see that same option being useful for template missions, too.

Current Method:
Template file missions which have basic events, skyboxes, etc setup are very useful, especially for mods. Right now the manual method for this process is:
1) Create/save a template mission 
2) Open it by browsing to it's location
3) Be very careful to click save to make a quick copy

Request:
The current method is a bit cumbersome and slow, and it could be very useful to have the following options in FRED:
1) Add an extra aspect to `File->New->` which includes a menu of templates. 
2) Make FRED look in `/data/templates` of the current mod and apply some `do not save without confirmation` behavior for files loaded from that location. In other words, if you open a template and then click save it will instead use `Save As` behavior (similar to when you create a new mission and try and save it).

**Comment by EatThePath** [comment]
@wookieejedi would it be useful as an intermediate measure to be able to define a template mission in your mod table that FRED always opens with? It doesn't solve everything if you have a bunch of different templates you want to pick from, but I know it'd be useful for me at least in clearing out some cruft from the initial file state.

**Comment by wookieejedi** [comment]
Yeah I reckon that'd be useful (saves me the step of opening the template mission every time FRED opens). The shortcut I added with Ctrl-Shift-S saves some time in that regard, but having an actual template open would be the most elegant solution. Remind me once stable is done to take a look at it.

**Comment by Goober5000** [comment]
If you implement this, make sure to also set some sort of flag where FRED should always prompt you with the file dialog the first time you save it.  Otherwise it would be very easy to overwrite the template mission.

**Comment by JohnAFernandez** [comment]
It would probably be better to have a separate file extension and a separate menu item to save as a template.  Something like "Save as template" and an extension of `.fsmt`.  Every time you would save as a template, it would have you go through the "save as" dialogue.
